flink-table-store-connector/src/main/java/org/apache/flink/table/store/connector/TableStoreFactory.java

```diff
@@ -309,4 +360,41 @@ TableStore buildTableStore(Context context) {
 
         return store;
     }
+
+    private Map<BinaryRowData, Map<Integer, List<DataFileMeta>>> pickManifest(
+            Map<BinaryRowData, Map<Integer, List<DataFileMeta>>> groupBy,
+            MergeTreeOptions options,
+            List<LogicalType> keyTypes) {
+        Map<BinaryRowData, Map<Integer, List<DataFileMeta>>> filteredGroupBy = new HashMap<>();
+        Comparator<RowData> keyComparator =
+                CodeGenUtils.generateRecordComparator(new TableConfig(), keyTypes, "KeyComparator")
+                        .newInstance(Thread.currentThread().getContextClassLoader());
```

Comment:
Let's make this into a separate method in a utility class ?

Reply:
> Let's make this into a separate method in a utility class ?

Yes, it has some duplicate logic here. Also, I was wondering if `FileStoreImpl` could expose `newKeyComparator`.